src/blockchain/deposit_strategy/deposit_transaction_sender.py

```diff
@@ -28,50 +42,115 @@ def _prepare_signs_for_deposit(quorum: list[DepositMessage]) -> tuple[tuple[str,
 
     def prepare_and_send(
         self,
+        module_id: int,
         quorum: list[DepositMessage],
         with_flashbots: bool,
-        is_mellow: bool,
     ) -> bool:
-        tx = self._prepare_mellow_tx(quorum) if is_mellow else self._prepare_general_tx(quorum)
-        return self._send_transaction(tx, with_flashbots)
+        if self._sender_checks(module_id):
+            tx = self._prepare_tx(quorum)
+            if tx and self._send_transaction(tx, with_flashbots):
+                return True
+
+        if self._next_sender is not None:
+            return self._next_sender.prepare_and_send(module_id, quorum, with_flashbots)
+
+        return False
 
-    def _prepare_mellow_tx(self, quorum: list[DepositMessage]) -> ContractFunction:
+    def _prepare_tx(self, quorum: list[DepositMessage]) -> Optional[ContractFunction]:
```

Comment:
Not good idea that class can be optional. Looks like we should expect `ContractFunction` or exception here

Reply:
Removed